<launch>
    <arg name="walls_scale" default="0.1" />

    <!-- params -->
    <param name="forest_file" value="$(find leg_tracker)/config/trained_leg_detector_res=0.33.yaml" />
    <param name="scan_topic" value="/scan_filtered" />
    <param name="fixed_frame" value="map"/>
    <param name="scan_frequency" value="7.5"/>

    <!-- run detect_leg_clusters -->
    <node pkg="leg_tracker" type="detect_leg_clusters" name="detect_leg_clusters" output="screen"/>

    <!-- run joint_leg_tracker -->
    <node pkg="leg_tracker" type="joint_leg_tracker.py" name="joint_leg_tracker" output="screen"/>

    <!-- run local_occupancy_grid_mapping -->
    <node pkg="leg_tracker" type="local_occupancy_grid_mapping" name="local_occupancy_grid_mapping" output="screen"/>

    <node pkg="social_nav_tests" type="leg_tracker_pedsim_publisher.py" name="leg_pedsim_publisher" output="screen" />

    <!-- <node pkg="pedsim_simulator" type="pedsim_frames_publisher.py" name="pedsim_frames_publisher" /> -->

    <arg name="hz" default="25.0" />

    <node name="pedsim_visualizer" type="pedsim_visualizer_node" pkg="pedsim_visualizer" output="screen">
        <param name="hz" value="$(arg hz)" type="double" />
        <param name="walls_marker_scale" value="$(arg walls_scale)" type="double" />
        <remap from="/pedsim_simulator/simulated_agents" to="/pedsim_simulator/simulated_agents" />
    </node>
</launch>